src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1633,12 +1633,16 @@ private NonReentrantLock StateLock
         ///
         /// This not intended to be the public API, use Document.WithFrozenPartialSemantics() instead.
         /// </summary>
-        public SolutionState WithFrozenPartialCompilationIncludingSpecificDocument(DocumentId documentId, CancellationToken cancellationToken)
+        public async ValueTask<SolutionState> WithFrozenPartialCompilationIncludingSpecificDocumentAsync(DocumentId documentId, CancellationToken cancellationToken)
         {
             try
             {
+                // Ensure we explicitly move to the BG so that if we do any IO it does not potentially happen
+                // on a UI thread caller.
+                await Task.Yield().ConfigureAwait(false);
```

Comment:
This is not necessary, and causes harm to the fast path. `GetSyntaxTreeAsync` is already asynchronous on the I/O paths.

Reply:
good point.  I'm unsure what the best thing is to do here.  The issue is lower down (in the lock) with FreezePartialStateWithTree.  that's the part that can cause IO, and i'm trying to ensure we're not on the UI thread there.  thoughts on a better approach?

Reply:
@CyrusNajmabadi: so are we able to defer the actual creation of the compilation and fetching of symbols a bit further to the actual GetCompilationAsync() call? I think that's the right fix here.